stellargraph/core/graph.py

```diff
@@ -778,20 +937,122 @@ def create_graph_schema(self, create_type_maps=True, nodes=None):
                         node_types[node_type_map[edge[1]]],
                     )
                 )
-                for edge in self.edges(keys=True, data=True)
+                for edge in self._graph.edges(keys=True, data=True)
             }
 
             gs.node_type_map = node_type_map
             gs.edge_type_map = edge_type_map
 
         return gs
 
+    ######################################################################
+    # Generic graph interface:
+
+    def is_directed(self) -> bool:
+        return self._graph.is_directed()
+
+    def number_of_nodes(self) -> int:
+        return self._graph.number_of_nodes()
+
+    def number_of_edges(self) -> int:
+        return self._graph.number_of_edges()
+
+    def nodes(self) -> Iterable[Any]:
```

Comment:
We have `list(G.nodes())` everywhere we need to get the nodes, as NetworkX returns an iterator.  Can we just convert to a list in this method, rather than in most calls to it? This can be part of the API.

Reply:
The answer is two-fold: 1. Always converting to a list is potentially a waste for a big graph if all we want to do is iterate over the nodes. 2. If we have a complex data-type (e.g. a node-type to node-data mapping), using `itertools.chain()` is very convenient (see #520).

Reply:
This is also an example of the design principle to prefer abstract types (e.g. Iterable) over concrete implementations (e.g. list). Part of the 'D' in 'SOLID', I think.

Reply:
Agreed. Although somehow we always end up converting to a list!

Reply:
Usually because we want to sub-sample the nodes. But I think we occasionally just iterate through the nodes.